### PR TITLE
DTSBPS-451: Integrating PaymentMessageProcessor with bulk scan processor client

### PR DESCRIPTION
**Pull request by SomeshRA**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSBPS-451

### Change description ###
Integrating PaymentMessageProcessor with bulk scan processor client

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/functionalTest/resources/application.yaml [hunk 1]
```diff
@@ -16,8 +16,9 @@ azure:
 
 bulk-scan-procesor:
   api:
-    retries: 10
-    wait-time-in-ms: 1000
+    url: ${PROCESSOR_URL}
+    retries: 5
+    wait-time-in-ms: 10
```
Comment (mdayican) [inline src/functionalTest/resources/application.yaml:21]: why do we need this?

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            new PaymentInfo("11234"),
+            new PaymentInfo("22234"),
+            new PaymentInfo("33234")
+        );
+
+        when(authTokenGenerator.generate()).thenReturn("authToken");
+
+        when(proxy.updateStatus(any(), any()))
+            .thenThrow(new HttpServerErrorException(GATEWAY_TIMEOUT, GATEWAY_TIMEOUT.getReasonPhrase(),
+                null, null, null))
```
Comment (mdayican) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java:71]: formatting

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            new PaymentInfo("11234"),
+            new PaymentInfo("22234"),
+            new PaymentInfo("33234")
+        );
+
+        when(authTokenGenerator.generate()).thenReturn("authToken");
+
+        when(proxy.updateStatus(any(), any()))
+            .thenThrow(new HttpServerErrorException(GATEWAY_TIMEOUT, GATEWAY_TIMEOUT.getReasonPhrase(),
+                null, null, null));
```
Comment (mdayican) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java:93]: formatting

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/BulkScanProcessorApiProxy.java [hunk 4]
```diff
@@ -14,7 +13,6 @@
 
 @FeignClient(name = "bulk-scan-processor-api", url = "${bulk-scan-procesor.api.url}",
     configuration = BulkScanProcessorApiProxy.BulkScanConfiguration.class)
-@Profile("!functional")
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/BulkScanProcessorApiProxy.java:17]: why removed?

File: src/functionalTest/resources/application.yaml (hunk 1 above)
Comment (SomeshRA) [reply src/functionalTest/resources/application.yaml]: Removed it after adding ignore functional profile.

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java (hunk 2 above)
Comment (SomeshRA) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java]: done.

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java (hunk 3 above)
Comment (SomeshRA) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java]: done

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/BulkScanProcessorApiProxy.java (hunk 4 above)
Comment (SomeshRA) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/BulkScanProcessorApiProxy.java]: Reinstate it after adding ignore functional profile to configuration files.

**Comment by apereverzin** [comment]
We have a policy to have config changes in separate PR(s). I suggest to extract all config changes from this PR to another PR.

File: Jenkinsfile_CNP [hunk 5]
```diff
@@ -90,7 +90,7 @@ withPipeline(type, product, component) {
   onNonPR() {
     loadVaultSecrets(nonPreviewSecrets)
   }
-  
+
```
Comment (mdayican) [inline Jenkinsfile_CNP:93]: pls remove

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+            new PaymentInfo("11234"),
+            new PaymentInfo("22234"),
+            new PaymentInfo("33234")
+        );
+
+        when(authTokenGenerator.generate()).thenReturn("authToken");
+
+        when(proxy.updateStatus(any(), any()))
+            .thenThrow(new HttpServerErrorException(GATEWAY_TIMEOUT, GATEWAY_TIMEOUT.getReasonPhrase(),
+             null, null, null))
```
Comment (mdayican) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java:71]: formatting

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+            new PaymentInfo("11234"),
+            new PaymentInfo("22234"),
+            new PaymentInfo("33234")
+        );
+
+        when(authTokenGenerator.generate()).thenReturn("authToken");
+
+        when(proxy.updateStatus(any(), any()))
+            .thenThrow(new HttpServerErrorException(GATEWAY_TIMEOUT, GATEWAY_TIMEOUT.getReasonPhrase(),
+             null, null, null));
```
Comment (mdayican) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java:93]: formatting

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+            new PaymentInfo("11234"),
+            new PaymentInfo("22234"),
+            new PaymentInfo("33234")
+        );
+
+        when(authTokenGenerator.generate()).thenReturn("authToken");
+
+        when(proxy.updateStatus(any(), any()))
+            .thenThrow(new HttpClientErrorException(BAD_REQUEST, BAD_REQUEST.getReasonPhrase(),
+             null, null, null));
```
Comment (mdayican) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java:117]: formatting

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClient.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+@Service
+@Profile("!functional")
+public class ProcessorClient {
+    private static final Logger logger = LoggerFactory.getLogger(ProcessorClient.class);
+    private final AuthTokenGenerator authTokenGenerator;
+    private final BulkScanProcessorApiProxy proxy;
+    private final RetryTemplate retryTemplate;
+
+    public ProcessorClient(AuthTokenGenerator authTokenGenerator, BulkScanProcessorApiProxy proxy,
+                           RetryTemplate retryTemplate) {
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClient.java:25]: formatting

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+    private ArgumentCaptor<PaymentRequest> paymentRequestArgumentCaptor;
+
+    private ProcessorClient processorClient;
+
+
+    @BeforeEach
+    void setUp() {
+        RetryConfig retryConfig = new RetryConfig();
+        processorClient = new ProcessorClient(authTokenGenerator, proxy,
+            retryConfig.retryTemplate(5, 500));
```
Comment (mdayican) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java:50]: fomatting

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+
+        List<PaymentInfo> paymentInfoList = of(
+            new PaymentInfo("11234"),
+            new PaymentInfo("22234"),
+            new PaymentInfo("33234")
+        );
+
+        when(proxy.updateStatus(any(), any()))
+            .thenThrow(new HttpServerErrorException(GATEWAY_TIMEOUT, GATEWAY_TIMEOUT.getReasonPhrase(),
+                null, null, null))
```
Comment (mdayican) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java:85]: formatting

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+
+        List<PaymentInfo> paymentInfoList = of(
+            new PaymentInfo("11234"),
+            new PaymentInfo("22234"),
+            new PaymentInfo("33234")
+        );
+
+        when(proxy.updateStatus(any(), any()))
+            .thenThrow(new HttpClientErrorException(UNAUTHORIZED, UNAUTHORIZED.getReasonPhrase(),
+                null, null, null))
```
Comment (mdayican) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java:110]: formatting

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+            .thenThrow(new HttpClientErrorException(BAD_REQUEST, BAD_REQUEST.getReasonPhrase(),
+                null, null, null));
```
Comment (mdayican) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/ProcessorClientTest.java:112]: formatting

**Pull request closed without merging**